vsdownload.py

```diff
@@ -88,12 +97,34 @@ def setPackageSelectionMSVC15(args, packages, userversion, sdk, toolversion, def
         args.package.extend(defaultPackages)
 
 def setPackageSelection(args, packages):
+    if not args.architecture:
+        if args.host_arch is not None:
+            args.architecture = [ args.host_arch ]
+        else:
+            args.architecture = ["x86", "x64", "arm", "arm64"]
+
     # If no packages are selected, install these versionless packages, which
     # gives the latest/recommended version for the current manifest.
-    extraarchs = ["ARM", "ARM64"]
-    defaultPackages = ["Microsoft.VisualStudio.Workload.VCTools", "Microsoft.VisualStudio.Component.VC.ATL"]
-    for arch in extraarchs:
-        defaultPackages.extend(["Microsoft.VisualStudio.Component.VC.Tools." + arch, "Microsoft.VisualStudio.Component.VC.ATL." + arch])
+    defaultPackages = []
+    if "x86" in args.architecture or "x64" in args.architecture:
+        defaultPackages.append("Microsoft.VisualStudio.Component.VC.Tools.x86.x64")
+        defaultPackages.append("Microsoft.VisualCpp.ASAN.X86")
+        defaultPackages.append("Microsoft.VisualStudio.Component.VC.ATL")
+    if "arm" in args.architecture:
+        defaultPackages.append("Microsoft.VisualStudio.Component.VC.Tools.ARM")
+        defaultPackages.append("Microsoft.VisualStudio.Component.VC.ATL.ARM")
+    if "arm64" in args.architecture:
+        defaultPackages.append("Microsoft.VisualStudio.Component.VC.Tools.ARM64")
+        defaultPackages.append("Microsoft.VisualStudio.Component.VC.ATL.ARM64")
+    
+    # TODO
+    defaultPackages.append("Win11SDK_10.0.22621")
```

Comment:
Is this also needed to compensate for missing dependencies when not using `Microsoft.VisualStudio.Workload.VCTools`?

If it turns out to be complicated to handle these bits, we could just unconditionally keep installing `Microsoft.VisualStudio.Workload.VCTools`, for now, and land the rest of the unproblematic parts of this PR, and then work on making more of the x86 tools optional in a separate later PR?